synthtool/gcp/common.py

```diff
@@ -321,11 +321,32 @@ def py_library(self, **kwargs) -> Path:
                     f.write(content)
         return ret
 
-    def java_library(self, **kwargs) -> Path:
+    def java_library(self, partial_files: List[str] = [], **kwargs) -> Path:
         # kwargs["metadata"] is required to load values from .repo-metadata.json
         if "metadata" not in kwargs:
             kwargs["metadata"] = {}
-        return self._generic_library("java_library", **kwargs)
+        # load common repo meta information (metadata that's not language specific).
+        if "metadata" in kwargs:
+            self._load_generic_metadata(
+                kwargs["metadata"],
+                partial_files=partial_files,
+            )
+            # if no samples were found, don't attempt to render a
+            # samples/README.md.
+            if "samples" not in kwargs["metadata"] or not kwargs["metadata"]["samples"]:
+                self.excludes.append("samples/README.md")
+
+        t = templates.TemplateGroup(self._template_root / "java_library", self.excludes)
+
+        if "repository" in kwargs["metadata"] and "repo" in kwargs["metadata"]:
+            kwargs["metadata"]["repo"]["default_branch"] = _get_default_branch_name(
+                kwargs["metadata"]["repository"]
+            )
+
+        result = t.render(**kwargs)
+        _tracked_paths.add(result)
+
```

Comment:
Can this logic go into the `java.common_templates()` call instead?

Context: when this code was originally written, the inheritance was backwards (this common template code had methods for each language and they were explicitly used in synth.py files in each repository). The intention was to migrate them all to use `<lang>.common_templates()` or something like that and then remove these language-specific methods on this class.

Reply:
I decided not to change the interface of `java.common_templates` and `java_library` as this refactoring should go to a dedicated PR.